### ISSUE TITLE
[Bug]: reference_only error with inpainting models

### ISSUE THREAD

**Issue by ThereforeGames** [Mikubill/sd-webui-controlnet#1268]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Hi, I'm receiving a crash with the new reference_only preprocessor when an Inpainting model is selected. Log attached.

### Steps to reproduce the problem

1. Select a model based on SD 1.5 Inpainting
2. Enable reference_only preprocessor
3. Generate img2img

### What should have happened?

img2img inference with reference_only

### Commit where the problem happens

webui: 1.2.0
controlnet: 1.1.166

### Command Line Arguments

```Shell
--xformers
```

### Console logs

```Shell
File "T:\code\python\automatic-stable-diffusion-webui\modules\call_queue.py", line 57, in f
    res = list(func(*args, **kwargs))
  File "T:\code\python\automatic-stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\modules\img2img.py", line 182, in img2img
    processed = process_images(p)
  File "T:\code\python\automatic-stable-diffusion-webui\modules\processing.py", line 526, in process_images
    res = process_images_inner(p)
  File "T:\code\python\automatic-stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\batch_hijack.py", line 42, in processing_process_images_hijack
    return getattr(processing, '__controlnet_original_process_images_inner')(p, *args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\modules\processing.py", line 680, in process_images_inner
    samples_ddim = p.sample(conditioning=c, unconditional_conditioning=uc, seeds=seeds, subseeds=subseeds, subseed_strength=p.subseed_strength, prompts=prompts)
  File "T:\code\python\automatic-stable-diffusion-webui\modules\processing.py", line 1135, in sample
    samples = self.sampler.sample_img2img(self, self.init_latent, x, conditioning, unconditional_conditioning, image_conditioning=self.image_conditioning)
  File "T:\code\python\automatic-stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 350, in sample_img2img
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "T:\code\python\automatic-stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 251, in launch_sampling
    return func()
  File "T:\code\python\automatic-stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 350, in <lambda>
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "T:\code\python\automatic-stable-diffusion-webui\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\repositories\k-diffusion\k_diffusion\sampling.py", line 145, in sample_euler_ancestral
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "T:\code\python\automatic-stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\modules\sd_samplers_kdiffusion.py", line 135, in forward
    x_out = self.inner_model(x_in, sigma_in, cond=make_condition_dict([cond_in], image_cond_in))
  File "T:\code\python\automatic-stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 112, in forward
    eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\repositories\k-diffusion\k_diffusion\external.py", line 138, in get_eps
    return self.inner_model.apply_model(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\modules\sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "T:\code\python\automatic-stable-diffusion-webui\modules\sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 858, in apply_model
    x_recon = self.model(x_noisy, t, **cond)
  File "T:\code\python\automatic-stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\repositories\stable-diffusion-stability-ai\ldm\models\diffusion\ddpm.py", line 1339, in forward
    out = self.diffusion_model(xc, t, context=cc)
  File "T:\code\python\automatic-stable-diffusion-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\hook.py", line 402, in forward_webui
    return forward(*args, **kwargs)
  File "T:\code\python\automatic-stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\hook.py", line 359, in forward
    ref_xt = ref_cond_xt * uc_mask + ref_uncond_xt * (1 - uc_mask)
RuntimeError: The size of tensor a (4) must match the size of tensor b (9) at non-singleton dimension 1
```

**Comment by ManOrMonster** [comment]
I'm trying to use it with an inpainting model in the SD-CN-Animation extension and received this error as well. Also with reference only.

**Comment by lllyasviel** [comment]
fixed in 1.1.170

**Comment by DenkingOfficial** [comment]
@lllyasviel Getting almost the same error with v1.1.204 but a is (9), b is (4)
```RuntimeError: The size of tensor a (9) must match the size of tensor b (4) at non-singleton dimension 1```

Model used: 1.5-inpainting
With not inpainting-based models it works perfectly fine

Full log:

```
16:00:50-064864 ERROR    Exception: The size of tensor a (9) must match the size of tensor b (4) at non-singleton
                         dimension 1
16:00:50-066839 ERROR    Arguments: args=('task(z690q8m5a4z8jrb)', '', '', [], 20, 0, False, False, 1, 1, 7, -1.0, -1.0,
                         0, 0, 0, False, 1024, 1024, True, 0.2, 1, 'Latent', 0, 0, 0, [], 0, False, 'MultiDiffusion',
                         False, True, 1024, 1024, 96, 96, 48, 4, 'None', 2, False, 10, 1, 1, 64, False, False, False,
                         False, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2,
                         '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0,
                         False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '',
                         'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False,
                         0.4, 0.4, 0.2, 0.2, '', '', 'Background', 0.2, -1.0, False, 0.4, 0.4, 0.2, 0.2, '', '',
                         'Background', 0.2, -1.0, True, 1024, 96, False, False, False, False, False, 7, 100, 'Constant',
                         0, 'Constant', 0, 4, False, False, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None',
                         1, 1, 'LoRA', 'None', 1, 1, 'LoRA', 'None', 1, 1, None, 'Refresh models',
                         <controlnet.py.UiControlNetUnit object at 0x000001BAE7FBB250>, False, False, 'Matrix',
                         'Horizontal', 'Mask', 'Prompt', '1,1', '0.2', False, False, False, 'Attention', False, '0',
                         '0', '0.4', None, False, '1:1,1:2,1:2', '0:0,0:0,0:1', '0.2,0.8,0.8', 150, 0.2, 1, 1, 1, 1, 1,
                         1, 1, 1, 1, 1, 1, 1, 'from modules.processing import process_images\n\np.width = 768\np.height
                         = 768\np.batch_size = 2\np.steps = 10\n\nreturn process_images(p)', 2, False, False,
                         'positive', 'comma', 0, False, False, '', 1, '', [], 0, '', [], 0, '', [], True, False, False,
                         False, 0, False, None, None, False, 50) kwargs={}
16:00:50-075840 ERROR    gradio call: RuntimeError
╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ C:\Codding\pure-sd-webui\modules\call_queue.py:34 in f                                                               │
│                                                                                                                      │
│    33 │   │   │   try:                                                                                               │
│ ❱  34 │   │   │   │   res = func(*args, **kwargs)                                                                    │
│    35 │   │   │   │   progress.record_results(id_task, res)                                                          │
│                                                                                                                      │
│ C:\Codding\pure-sd-webui\modules\txt2img.py:56 in txt2img                                                            │
│                                                                                                                      │
│   55 │   if processed is None:                                                                                       │
│ ❱ 56 │   │   processed = process_images(p)                                                                           │
│   57 │   p.close()                                                                                                   │
│                                                                                                                      │
│                                               ... 22 frames hidden ...                                               │
│                                                                                                                      │
│ C:\Codding\pure-sd-webui\extensions\sd-webui-controlnet\scripts\hook.py:572 in forward                               │
│                                                                                                                      │
│   571 │   │   │   │   t = torch.round(timesteps.float()).long()                                                      │
│ ❱ 572 │   │   │   │   x0_prd = predict_start_from_noise(outer.sd_ldm, x, t, h)                                       │
│   573 │   │   │   │   x0 = x0_prd * mask + x0_origin * (1 - mask)                                                    │
│                                                                                                                      │
│ C:\Codding\pure-sd-webui\extensions\sd-webui-controlnet\scripts\hook.py:221 in predict_start_from_noise              │
│                                                                                                                      │
│   220 def predict_start_from_noise(ldm, x_t, t, noise):                                                              │
│ ❱ 221 │   return extract_into_tensor(ldm.sqrt_recip_alphas_cumprod, t, x_t.shape) * x_t - extr                       │
│   222                                                                                                                │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
RuntimeError: The size of tensor a (9) must match the size of tensor b (4) at non-singleton dimension 1
```
